cmd/promtool/main.go

```diff
@@ -368,6 +375,52 @@ func checkTLSConfig(tlsConfig config_util.TLSConfig) error {
 	return nil
 }
 
+func checkSDFile(filename string) error {
+	fd, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+
+	content, err := ioutil.ReadAll(fd)
+	if err != nil {
+		return err
+	}
+
+	var targetGroups []*targetgroup.Group
+
+	switch ext := filepath.Ext(filename); strings.ToLower(ext) {
+	case ".json":
+		if err := json.Unmarshal(content, &targetGroups); err != nil {
+			return err
+		}
+	case ".yml", ".yaml":
+		if err := yaml.UnmarshalStrict(content, &targetGroups); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("invalid file extension: %q", ext)
+	}
+
+	_, err = checkSDOutput(targetGroups)
+	return err
+}
+
+func checkSDOutput(targetGroups []*targetgroup.Group) ([]*targetgroup.Group, error) {
```

Comment:
Although the processed target groups aren't currently used, I wrote it anticipating adding a standalone command to check an SD output and display the result.

https://github.com/prometheus/prometheus/issues/8935#issuecomment-860730894